homeassistant/components/switchbee/coordinator.py

```diff
@@ -26,15 +61,36 @@ def __init__(
         """Initialize."""
         self.api: CentralUnitAPI = swb_api
         self._reconnect_counts: int = 0
-        self.mac_formated: str = format_mac(swb_api.mac)
+
+        self.mac_formated: str | None = (
+            format_mac(self.api.mac) if isinstance(self.api.mac, str) else None
+        )
+
```

Comment:
Same here, it is more efficient to check against `None`.
```suggestion

        self.mac_formated: str | None = (
            None if self.api.mac is None else format_mac(self.api.mac)
        )

```